nereid/nereid/tests/test_src/network/test_tasks.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+from nereid.src.network.tasks import (
+    validate_network,
+    network_subgraphs,
+    render_subgraph_svg,
+)
+
+
+@pytest.mark.parametrize(
+    "edgelist, isvalid, exp",
+    [
+        (
+            [("a", "b"), ("b", "c"), ("d", "c"), ("c", "e")],
+            True,
+            ([], [], [], []),
+        ),  # this one's valid
+        (
+            [("a", "b"), ("b", "c"), ("d", "c"), ("c", "a")],
+            False,
+            ([["a", "b", "c"]], [("a", "b", 0), ("b", "c", 0), ("c", "a", 0)], [], []),
+        ),  # cycle a->b->c->a
+        (
+            [("a", "b"), ("b", "c"), ("d", "c"), ("c", "e"), ("c", "e")],
+            False,
+            ([], [], [("c", 2)], [("c", "e")]),
+        ),  # duplicate edge
+        (
+            [("a", "b"), ("b", "c"), ("d", "c"), ("a", "e")],
+            False,
+            ([], [], [("a", 2)], []),
+        ),  # multiple out edges, a->b & a->e
+    ],
+)
+def test_validate_network(edgelist, isvalid, exp):
+
+    g = {"edges": [{"source": s, "target": t} for s, t in edgelist], "directed": True}
+
+    result = validate_network(g)
+
+    if isvalid:
```

Comment:
simply: 

```python
assert result["isvalid"] == isvalid
```
?

Reply:
the variable `isvalid` is passed to the test from the parametrize list, and indicates that the tester thinks it should be a valid network. the value stored in `result['isvalid']` is the returned response from the API, and should contain either True or False to match the test expectation.

Reply:
wait, i understand this comment now. these 4 line snips are too short. i'll make this change, and change the variable to 'expected' rather than 'isvalid'.